blog/2024-11-20-integration-quality-scale.md

```diff
@@ -0,0 +1,29 @@
+---
+author: Joost Lekkerkerker
+authorURL: https://github.com/joostlek
+authorImageURL: https://avatars.githubusercontent.com/u/7083755?v=4
+title: "Integration quality scale"
+---
+
+Since the original quality scale has been introduced in 2020, Home Assistant changed quite a lot.
+Both in how integrations are developed and what features they can provide.
+The quality scale, however, has not been updated to reflect these changes.
+
+The last couple of months we have been working with the community on a new integration quality scale, that does fit the current state of Home Assistant.
+To give an overview of the changes:
+
+- The quality scale now has four scaled tiers: Bronze, Silver, Gold, and Platinum. The newly added Bronze tier is the new minimum requirement for new core integrations going forward. This does lower the bar for new integrations for a bit, but it does make it clear upfront what is expected from a core integration in this day and age.
+- A new non-scaled tier, legacy, has been added for integration that are not configurable via the UI yet. This tier would indicate to the user that the integration might not meet their expectation compared to integrations configurable via the UI.
+- To allow the scale to grow with Home Assistant and to avoid another major rework in the future, we have designed a rulebook for the scale so it's known what needs to happen when a rule is added.
+- To show that the project has a commitment on keeping the scale up to date, we have created an architecture decision record (ADR) that describes the goal and the process of keeping the scale up to date.
+- The developer documentation now has a brand new section dedicated to the Integration Quality Scale. It includes the rules, the reasoning for every rule and it provides examples that can be used to give an idea of how to implement it. There is also a checklist generated that can be used for changing scales.
+- Documentation now also plays a role in the quality of the integrations, since the user experience of an integration goes beyond code.
+- To track the progress of integration, we have added a `quality_scale.yaml` file to the integration folder. This will contain a list of all the rules it is adhering to and notes down possible exemptions and the reason behind the exemption.
```

Comment:
```suggestion
- To track the progress of integration, we have added [a `quality_scale.yaml` file](https://github.com/home-assistant/core/blob/dev/homeassistant/components/airgradient/quality_scale.yaml) to the integration folder. This will contain a list of all the rules it is adhering to and notes down possible exemptions and the reason behind the exemption.
```